sys/Makefile.dep

```diff
@@ -31,6 +31,12 @@ ifneq (,$(filter crypto_%,$(USEMODULE)))
   USEMODULE += crypto
 endif
 
+ifneq (,$(filter ieee802154_security,$(USEMODULE)))
+  USEMODULE += crypto
+  USEMODULE += crypto_aes
+  USEMODULE += cipher_modes
+endif
+
```

Comment:
If you have one transceiver with hardware support and one that does not, all of them are used.
If you only have transceivers with hardware support, I think only `crypto` and `crypto_aes` are used, because I call `cipher_init()` in `ieee802154_sec_init` and the inclusion of `crypto_aes` determines the size of
`cipher_t::cipher_context_t context` to fit for the maximum AES key size. If I only had transceivers with hardware support, I would only need a `uint8_t[16]` to store the key, instead of a `cipher_t` inside of `ieee802154_sec_context_t`.

But distinguishing between the cases `at least one transceiver has no hardware support` and `all transceivers have hardware support` smells like ugly `#if` , `#else`, so i decided to just always have a `cipher_t`. But I could probably do
```
ifeq (,$(filter at86rf2xx_aes_spi,$(USEMODULE)))
  USEMODULE += cipher_modes
endif
```

Or I could create a pseudomodule `IEEE802154_SECURITY_HW_ONLY` which must be added by the user, if he knows that all transceivers have hardware support. Then I´d modify
```C
 struct ieee802154_sec_context {
#if IS_USED(IEEE802154_SECURITY_HW_ONLY)
    uint8_t key[16];
#else
     cipher_t cipher;
#endif
 ...
```

and create a macro to access the key
```C
#if IS_USED(IEEE802154_SECURITY_HW_ONLY)
    #define IEEE802154_SECURITY_ACCESS_KEY(ctx) (ctx->key);
#else
    #define IEEE802154_SECURITY_ACCESS_KEY(ctx) (ctx->cipher.context.context)
#endif
```
Do you think the `IEEE802154_SECURITY_HW_ONLY` pseudomodule would be better?
I think it would be nice actually.

Reply:
> But distinguishing between the cases at least one transceiver has no hardware support and all transceivers have hardware support smells like ugly #if , #else, so i decided to just always have a cipher_t.

:+1:

Reply:
Maybe this is something to look into as follow up to not let the complexity of this PR increase more than needed.